client-mr/core/src/main/java/org/apache/hadoop/mapreduce/task/reduce/RssEventFetcher.java

```diff
@@ -75,7 +77,11 @@ public Roaring64NavigableMap fetchAllRssTaskIds() {
     String errMsg = "TaskAttemptIDs are inconsistent with map tasks";
     for (TaskAttemptID taskAttemptID : successMaps) {
       if (!obsoleteMaps.contains(taskAttemptID)) {
-        long rssTaskId = RssMRUtils.convertTaskAttemptIdToLong(taskAttemptID, appAttemptId);
+        int maxFailures = jobConf.getInt(MRJobConfig.MAP_MAX_ATTEMPTS, 4);
+        boolean speculation = jobConf.getBoolean(MRJobConfig.MAP_SPECULATIVE, true);
```

Comment:
Default values for `MAP_MAX_ATTEMPTS` and `MAP_SPECULATIVE` defined in at least two places, here and client-mr/core/src/main/java/org/apache/hadoop/mapred/RssMapOutputCollector.java. With #1427, default values will be defined once.